S4.Rmd

```diff
@@ -190,6 +217,8 @@ Note that this example uses another form of `new()`: unnnamed arguments are take
 
 S4 allows you to define a "validity" method that checks if your object is valid. You set this method by calling `setValidity()`. Here's a port of our factor validity checking to S4. A validity method should return a character vector containing a list of all of the problems. If there are no problems, it should return a character vector of length 0.
 
+<!-- FIXME: generally, one would like to discourage direct slot access, providing accessors (e.g., plain-old-functions) instead, and using these throughout, including in validateFactor -->
```

Comment:
you're probably your own most frequent user, at least initially and when the implementation is most fluid, so benefit the most from separating interface from implementation.